### ISSUE TITLE
Fetch funders from KG and allow user to select existing funders when creating funding instances

### ISSUE THREAD

**Issue by ehennestad** [HumanBrainProject/ebrains_wizard#11]
- [x] Fetch funders from KG
- [x] Update form for creating funding
- [ ] Update json format to store all necessary funding properties

**Comment by lzehl** [comment]
Be careful with semantics here. There is no schema in openMINDS for "funder" there is only a schema for "funding" and a property called "funder". There is a difference because "funders" can fund multiple causes, while a funding is specifically granted for one cause (which can of course have multiple products. A "funder" is a role that a Consortium, Organization or Person has that Consortium, Organization or Person provides funding for something.

That means you can fetch multiple things here:
+ fetch whole funding instances (e.g., HBP SGA1)
+ fetch Consortium, Organization or Person for a newly created funding instances for the property "funder" (e.g. EU)

**Comment by ehennestad** [comment]
Yes, my current thinking is to fetch Person, Organisation and Consortium, and letting the user select one of those in addition to entering award number and award title.

**Comment by lzehl** [comment]
Okay. This will lead to duplication in the EBRAINS KG for funding instances. I would suggest to provide both options, or it is necessary for the curation team afterwards to identify and eliminate duplicates before an upload to the EBRAINS KG.

**Comment by ehennestad** [comment]
Yes, this is also an option!

I don't have a sense of how often a funding instance is reused, i.e how many datasets are typically tied to the same funding instance. My assumption is that this is the exception, but if I am wrong it makes sense to include an option to select a funding instance or create a new one.

My second concern is how to present funding instances. I have seen many cases where award numbers and/or award titles are missing. I feel such a list would appear very difficult to navigate, although with search and if you know exactly what you are looking for it could be helpful.

As for duplications, I assume the award number will always be unique. It's always possible to search for duplicate award numbers programmatically and flag these for the curators. This will not catch typos or other spurious ways data can be entered, but that should also be an exception I hope

**Comment by lzehl** [comment]
Well during HBP all datasets that received funding through HBP had the same funding instance(s). The same will happen for EBRAINS 2.0 and other bigger projects

And even on a smaller scale: datasets from the same lab or institute also often have overlapping funding instance(s).

So I don't think this is an exception.

**Comment by lzehl** [comment]
To the quality of the funding instances. This is a curation issue within EBRAINS that has to be tackled. Even if openMINDS does not require it, EBRAINS should require at least the award title.

**Comment by lzehl** [comment]
As for the award numbers: they should be unique within the system of a funder. they are not globally unique. AND they are often written by data providers in multiple ways (e.g. HBP SGA2 No. 785907 VS. EU Specific Grant Agreement No. 785907 VS. SGA No. 785907 etc.).

**Comment by lzehl** [comment]
What you could do for a UI:

1) start asking for a funder (fetch Consortium, Organization or Person; probably with Organization as default)

2) provide list of existing Funding instances connected to this funder (which is a small query). Consolidate info of Funding instance in UI to one string.

Preparation for this by the Curation Team: Cleanup of Funding instances (and Consortium, Organization and Person instances).
